stream.go

```diff
@@ -172,23 +186,38 @@ func (cs *clientStream) RecvMsg(m interface{}) (err error) {
 			return toRPCErr(errors.New("grpc: client streaming protocol violation: get <nil>, want <EOF>"))
 		}
 		if err == io.EOF {
+			if EnableTracing {
+				defer cs.traceInfo.tr.Finish()
```

Comment:
why is this buried all the way in here? This seems like it'll be missed in several situations (e.g. the error handling above here), and the scope of this method is not the same as the scope of the RPC.